### ISSUE TITLE
Provide a way to not generate extensions with ALT_SOABI

### ISSUE THREAD

**Issue by lu-zero** [PyO3/maturin#1430]
I'm trying to use orjson on aarch64-openwrt-linux-musl.

The extension fails to load because the provided python only looks for the SOABI: `cpython-310`.

Might be a good idea to provide the SOABI variant when the wheel is built with --release.

**Comment by messense** [comment]
How did you build the extension for `aarch64-openwrt-linux-musl`, is it cross compiled?

`orjson` uses `pyo3` so you can already change it via a `PYO3_CONFIG_FILE`: https://pyo3.rs/v0.18.0/building_and_distribution#advanced-config-files, use the (maturin specific) `ext_suffix` option to change the `.so` filename, for example

```
implementation=CPython
version=3.10
pointer_width=64
ext_suffix=.cpython-310.so
```

**Comment by lu-zero** [comment]
Yesterday I ended up trying to figure out what openwrt is doing, so I'm asking them to drop the patch that causes the problem on their side. The extension has to be cross compiled and if they won't drop that patch your solution seem good :)

**Comment by lu-zero** [comment]
Does not seem to work  though.
```
  --- stderr
  error: failed to parse contents of PYO3_CONFIG_FILE
  caused by:
    - 0: unknown config key `ext_suffix`
```

**Comment by messense** [comment]
Ok, I think it's because currently pyo3-build-config is being too strict about unknown config keys:

https://github.com/PyO3/pyo3/blob/efb8a1241402bbad03b341bc2d8ce2613596a793/pyo3-build-config/src/impl_.rs#L436

**Comment by messense** [comment]
> Ok, I think it's because currently pyo3-build-config is being too strict about unknown config keys:
> 
> https://github.com/PyO3/pyo3/blob/efb8a1241402bbad03b341bc2d8ce2613596a793/pyo3-build-config/src/impl_.rs#L436

@davidhewitt What do you think about changing from denying to warn about unknown config keys in `pyo3-build-config`? Otherwise in maturin we need to copy the user supplied `PYO3_CONFIG_FILE` and remove the maturin specific config keys beforehand.

**Comment by davidhewitt** [comment]
Sure, changing to warn-by-default seems ok to me.

**Comment by messense** [comment]
> Sure, changing to warn-by-default seems ok to me.

https://github.com/PyO3/pyo3/pull/2926

**Comment by messense** [comment]
Resolved in `pyo3` 0.18.1: https://github.com/PyO3/pyo3/blob/0820c95cb545ea86f215424099df54e881f586f0/CHANGELOG.md#0181---2023-02-07
